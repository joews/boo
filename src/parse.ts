import { readFileSync } from 'fs'
import { join } from 'path'
import { generate } from 'pegjs'

const parserSource = readFileSync(join(__dirname, 'boo.pegjs'), 'utf8')
const parser = generate(parserSource)

export default function parse (source: string): any {
  const ast = parser.parse(source)
  // console.log(`TRACE parse: `, JSON.stringify(ast))
  return ast
}
