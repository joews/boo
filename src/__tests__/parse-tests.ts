/* eslint-env jest */
import parse from '../parse'

function testParse (source: string) {
  test(source, () => {
    // console.log(JSON.stringify(parse(source)))
    expect(parse(source)).toMatchSnapshot()
  })
}

testParse("iconst 3")
