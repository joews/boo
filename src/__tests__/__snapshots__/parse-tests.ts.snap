// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iconst 3 1`] = `
Array [
  Object {
    "mnemonic": "iconst",
    "operands": Array [
      3,
    ],
  },
]
`;
